core/DataAccess/LogAggregator.php

```diff
@@ -1084,67 +1100,92 @@ public function queryConversionsByDimension($dimensions = array(), $where = fals
-                    sprintf('%s AS `%d`', self::getSqlRevenue('log_conversion.revenue_subtotal'), Metrics::INDEX_GOAL_ECOMMERCE_REVENUE_SUBTOTAL),
-                    sprintf('%s AS `%d`', self::getSqlRevenue('log_conversion.revenue_tax'), Metrics::INDEX_GOAL_ECOMMERCE_REVENUE_TAX),
-                    sprintf('%s AS `%d`', self::getSqlRevenue('log_conversion.revenue_shipping'), Metrics::INDEX_GOAL_ECOMMERCE_REVENUE_SHIPPING),
-                    sprintf('%s AS `%d`', self::getSqlRevenue('log_conversion.revenue_discount'), Metrics::INDEX_GOAL_ECOMMERCE_REVENUE_DISCOUNT),
-                    sprintf('log_conversion.items AS `%d`', Metrics::INDEX_GOAL_ECOMMERCE_ITEMS),
-                    sprintf('1 AS `%s`', Metrics::INDEX_GOAL_NB_CONVERSIONS_PAGE_UNIQ),
-                )
-            ),
-            // FROM...
-            array(
-                self::LOG_CONVERSION_TABLE,
-                array(
-                    "table" => "log_link_visit_action",
-                    "joinOn" => "log_link_visit_action.idvisit = log_conversion.idvisit AND log_link_visit_action.server_time <= log_conversion.server_time AND log_link_visit_action.".$linkField." IS NOT NULL"
-                ),
-                array(
-                    "table" => "log_action",
-                    "joinOn" => "log_action.idaction = log_link_visit_action.".$linkField." AND ".($linkField == 'idaction_url' ? 'log_action.type = 1' : 'log_action.type = 4')
-                )
-            ),
-            // WHERE ... AND ...
-            implode(
-                ' AND ',
-                array(
-                    'log_conversion.server_time >= ?',
-                    'log_conversion.server_time <= ?',
-                    'log_conversion.idsite IN ('.Common::getSqlStringFieldsArray($this->sites).')',
-                    'log_conversion.idgoal >= 0'
-                )
-            ),
-
-            // GROUP BY ...
-            false,
-
-            // ORDER ...
-            'NULL'
-        );
-
-        return $this->getDb()->query($query['sql'], $query['bind']);
+        $dbSettings = new \Piwik\Db\Settings();
+        $tablePrefix = $dbSettings->getTablePrefix();
+
+        $localBind = $this->getGeneralQueryBindParams();
+        $bind[] = $localBind[0]; // Start date
+        $bind[] = $localBind[1]; // End date
+        $bind[] = $localBind[2]; // Sites
+        $bind[] = $localBind[0];
+        $bind[] = $localBind[1];
+        $bind[] = $localBind[2];
+
+        // This query is too complicated to be built by the generateQuery function
+        // All non-bound parameters are either typed numeric or supplied from fixed values in other methods (linkField, idGoal, tablePrefix)
+        // All user supplied parameters are bound (idSite,  date)
+
+        $sql = sprintf(
+        "SELECT
+          ".$idGoal." AS idgoal,
+          ".($linkField == 'idaction_url' ? '1' : '4')." AS `type`,
+          idaction AS idaction,
```

Comment:
In database queries we should try always use a table identifier for columns, to avoid possible problems with ambiguous field names.